### PR TITLE
README updates for Utils.obtainServerConfig()

### LINKED ISSUE #221: EXTJSDK: Update READMEs to reflect new environment variable enhancement
Need to update the READMEs of all the connectors as well as the `extjsdk` to reflect the new change that allows the authToken to be specified as an environment variable.

Most likely will update the `extjsdk` and then have the connector READMEs point to that top-level description.

### PR DESCRIPTION
**Pull request by namirfa**
Closes #221 

Added documentation for the `Utils.obtainServerConfig()` method in the extjsdk/README. As part of that documentation, added a description of the new enhancement to retrieve the `authToken` from an environment variable.

Each connector's README now references this documentation and encourages reading the SDK Server Config documentation first before reading the connector specific one.

### CODE REVIEW

File: extjsdk/README.md [hunk 1]
```diff
@@ -34,6 +34,25 @@ dependencies on mvnrepository.com are included.
...
+Generally, connectors need a minimum of three configuration properties at startup:
+*   `targetServer`: The URL of the Vantiq server to which the connector should connect.
+*   `authToken`: The access token that the connector will use to authenticate to the desired namespace.
+*   `source(s)`: The name of the source in the namespace (or in some cases a list of comma-separated source names) 
+    that the connector will connect to
+    
+The SDK includes a utility method to retrieve the startup configuration document and parse its contents into a 
+`Properties` object: `Utils.obtainServerConfig()`. By default, this method will look for those values in a file named 
+`server.config`. That file is expected to be included in the running connector directory as follows: 
+`<connectorRunningDirector>/serverConfig/server.config`. The SDK will also look for the file in the running connector
```
Comment (fhcarter) [inline extjsdk/README.md:47]: I believe that the _term of art_ is the  _connectors working directory_.  $CWD in scripts, etc., generally refers to Current Working Directory.

File: extjsdk/README.md [hunk 2]
```diff
@@ -34,6 +34,25 @@ dependencies on mvnrepository.com are included.
...
+## <a name="serverConfig" id="serverConfig"></a>Connector Startup Configuration
+Generally, connectors need a minimum of three configuration properties at startup:
+*   `targetServer`: The URL of the Vantiq server to which the connector should connect.
+*   `authToken`: The access token that the connector will use to authenticate to the desired namespace.
+*   `source(s)`: The name of the source in the namespace (or in some cases a list of comma-separated source names) 
+    that the connector will connect to
+    
+The SDK includes a utility method to retrieve the startup configuration document and parse its contents into a 
+`Properties` object: `Utils.obtainServerConfig()`. By default, this method will look for those values in a file named 
+`server.config`. That file is expected to be included in the running connector directory as follows:
```
Comment (fhcarter) [inline extjsdk/README.md:46]:
```suggestion
`server.config`. That file is expected to be included in the running connector's working directory as follows: 
```

File: extjsdk/README.md [hunk 3]
```diff
@@ -34,6 +34,25 @@ dependencies on mvnrepository.com are included.
...
+    
+The SDK includes a utility method to retrieve the startup configuration document and parse its contents into a 
+`Properties` object: `Utils.obtainServerConfig()`. By default, this method will look for those values in a file named 
+`server.config`. That file is expected to be included in the running connector directory as follows: 
+`<connectorRunningDirector>/serverConfig/server.config`. The SDK will also look for the file in the running connector 
+directory if it is not found in the `serverConfig` subdirectory (i.e. `<connectorRunningDirector>/server.config`).
+
+For users who may not want to write the `authToken` property to a file because of its sensitive nature, the 
+`Utils.obtainServerConfig()` method will also search for this value in an environment variable named 
+`CONNECTOR_AUTH_TOKEN`. If the `authToken` is specified in the `server.config` document, that value will take precedent.
```
Comment (fhcarter) [inline extjsdk/README.md:52]:
```suggestion
`CONNECTOR_AUTH_TOKEN`. If the `authToken` is specified in the `server.config` document, that value will take precedence.
```

File: CSVSource/README.md [hunk 4]
```diff
@@ -45,6 +45,8 @@ name for each class is the class's fully qualified class name, *e.g.* "io.vantiq
 
 ## Server Config File
 
+(Please read to the [SDK's server config documentation](../extjsdk/README.md#serverConfig) first.)
+
 The server config file is written as `property=value`, with each property on its
```
Comment (fhcarter) [inline CSVSource/README.md:50]: Seems like we could move all of this stuff (down thru `Setting up Vantiq`) to the extsdk readme, and then just refer folks there for the whole thing.  The less repetition, the better.

**Review by fhcarter — APPROVED** [review]
LG.  I'd suggest moving the entire description & example of the server.config file to the SDK, then delete it from the connectors.  For most connectors, then the server config section becomes "Please see the SDK's server config documentation...".

For cases where we still describe the deprecated stuff (UDP & OPCUA), it just gets finagled a bit, but still refers to the common case.

File: CSVSource/README.md [hunk 5]
```diff
@@ -44,16 +44,7 @@ which is an [Apache Log4j configuration file.](https://logging.apache.org/log4j/
...
-The server config file is written as `property=value`, with each property on its
-own line. 
-The following is an example of a valid server.config file:
-
-```
-authToken=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
-sources=CSV1
-targetServer=https://dev.vantiq.com/
-```
+(Please read to the [SDK's server config documentation](../extjsdk/README.md#serverConfig) first.)
```
Comment (rblumer4) [inline CSVSource/README.md:47]: same typo in most of these:  remove "to" so it is just:
`Please read the ...`

**Pull request merged**